flowy/src/lib.rs

```diff
@@ -176,24 +176,36 @@ pub fn set_times(config: Config) -> Result<(), Box<dyn Error>> {
     }
 }
 
-/// Returns the index of the wallpaper path which is
-/// closest to the current time
-fn get_current_wallpaper_idx(wall_len: &Vec<String>) -> Result<usize, Box<dyn Error>> {
+/// Returns the index of the wallpaper which should be displayed now.
+///
+/// For example, if the times are "00:00", "01:00" and "02:00", the first image
+/// should be shown from 00:00 to 00:59 and the second image from 01:00 to 01:59.
+///
+/// Therefore, this function returns the index of the _last_ time that isn't
+/// greater than the current time.
+fn get_current_wallpaper_idx(wall_times: &[String]) -> Result<usize, Box<dyn Error>> {
+    if wall_times.is_empty() {
+        panic!("Array of times can't be empty");
+    }
+
     // Get the current time
     let curr_time = Local::now().time();
-    let mut global_min = 1440;
+
+    // calculate index + 1
     let mut index = 0;
-    for (i, time) in wall_len.into_iter().enumerate() {
-        // Get the difference in absolute minutes
-        let time = chrono::NaiveTime::parse_from_str(&time, "%H:%M")?;
-        let min = curr_time.signed_duration_since(time).num_minutes().abs();
-        // Compare and see if lowest we have seen so far
-        if min < global_min {
-            global_min = min;
-            index = i;
+    for time in wall_times {
+        let time = NaiveTime::parse_from_str(&time, "%H:%M")?;
+        if time > curr_time {
+            break;
         }
+        index += 1;
+    }
+    if index == 0 {
+        // The first time should ALWAYS be "00:00". If not, this can happen:
```

Comment:
First time isn't always 00:00. You should take this into account.

Reply:
What's the expected behavior then? To display the last image between 00:00 and the first time?

Reply:
Expected behaviour is that if the closest time is the starting time, then basically set the wallpaper to the time before it (cyclically, this would be the last wallpaper)